src/cargo/ops/cargo_package/mod.rs

```diff
@@ -836,11 +837,12 @@ fn check_metadata(pkg: &Package, gctx: &GlobalContext) -> CargoResult<()> {
         }
         things.push_str(missing.last().unwrap());
 
-        gctx.shell().warn(&format!(
-            "manifest has no {things}.\n\
-             See https://doc.rust-lang.org/cargo/reference/manifest.html#package-metadata for more info.",
-            things = things
-        ))?
+        gctx.shell().print_report(&[
+            Level::WARNING.secondary_title(format!("manifest has no {things}."))
+                .element(Level::NOTE.message("See https://doc.rust-lang.org/cargo/reference/manifest.html#package-metadata for more info."))
```

Comment:
lowercase and no perild according to the [diagnostic style guide](https://rustc-dev-guide.rust-lang.org/diagnostics.html#diagnostic-output-style-guide)

```suggestion
                .element(Level::NOTE.message("see https://doc.rust-lang.org/cargo/reference/manifest.html#package-metadata for more info"))